packages/engine/lib/simulation_structure/src/lib.rs

```diff
@@ -1,45 +1,21 @@
-use core::fmt;
-use std::str::FromStr;
-
-use serde::{Deserialize, Serialize};
-use uuid::Uuid;
-
-pub type SimulationShortId = u32;
-
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(transparent)]
-pub struct ExperimentId {
-    id: Uuid,
-}
-
-impl ExperimentId {
-    pub fn generate() -> Self {
-        Self { id: Uuid::new_v4() }
-    }
-
-    pub fn as_bytes(&self) -> &[u8; core::mem::size_of::<u128>()] {
-        self.id.as_bytes()
-    }
-}
-
-impl fmt::Display for ExperimentId {
-    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.serialize(fmt)
-    }
-}
-
-impl From<ExperimentId> for Uuid {
-    fn from(id: ExperimentId) -> Self {
-        id.id
-    }
-}
-
-impl FromStr for ExperimentId {
-    type Err = uuid::Error;
-
-    fn from_str(uuid_str: &str) -> Result<Self, Self::Err> {
-        Ok(Self {
-            id: Uuid::from_str(uuid_str)?,
-        })
-    }
-}
+//! Contains the structure for running the engine.
+//!
+//! The configuration of an experiment is defined in [`Manifest`]. A [`Manifest`] contains the
+//! definition to run an [`Experiment`], which is described as [`ExperimentRun`]. An [`Experiment`]
+//! has a [`Simulation`] associated with it. Depending on the [`ExperimentType`] the [`Experiment`]
+//! will run the [`Simulation`] with different configurations.
```

Comment:
```suggestion
//! The [`Manifest`] defines the initial configuration used to define an experiment, this is then turned into an [`ExperimentRun`]. Within the [`ExperimentRun`] there are specific configurations for the [`Experiment`] and the [`Simulation`]s within the experiment. 
```

I've deleted the next line just because I don't know how to rephrase it at the moment, and it's a little confusing to link to the [`Simulation`] struct, which is actually just a config (it's not _a_ Simulation), when actually talking about simulations in general.

Can we just fix this and possibly include [`ExperimentType`] as part of the TODO below? :)

Reply:
That might be worthwhile too, can you add that to whatever TODO/task we have for thinking about renaming the crate

Reply:
Added it to the task linked in the OP